### PR TITLE
fix(compiler): fix the remaining EVM SSA stack-lift crashes

### PR DESCRIPTION
**Pull request by abmcar**
Fix the remaining crashes in the EVM multipass JIT path that lifts the operand stack into SSA form.

## What

Follow-up to #530 (now merged). Fixes the remaining crashes in the EVM
multipass JIT path that lifts the operand stack into SSA form
(`ZEN_ENABLE_EVM_STACK_SSA_LIFT`, default OFF). After this change, the lift
path compiles the full EEST state-test suite and the mainnet-replay corpus
crash-free, producing results identical to the default build. All edits are
reachable only on the lift path; the default (flag-off) build is unaffected.

## The bugs

1. **Dynamic-jump shape-class entry-depth gate** — a block that is both a
   dynamic-jump target and a dynamic-jump source whose own `JUMP` net-pops the
   stack (entry depth ≠ exit depth) had its shallow exit stack assigned as a
   deeper region's entry state → `EntryDepth == Values.size()` abort. Gate the
   shape-class fallback on the source's exit depth matching the class entry
   depth (`evm_analyzer.h`).

2. **JUMPDEST double-begin** — when a `JUMP`/`JUMPI` terminator falls through
   into a `JUMPDEST`, both the terminator handler and the main loop begin the
   same block. Benign with the flag OFF; with the lift path enabled it adds a
   self-edge `PredBlockPC == BlockPC` (→ `getPhiIncomingSlot` abort) and
   re-restores the lifted entry state (→ depth-mismatch abort). Detect the
   re-begin (`RunStartPC == CurrentBlockEntryPC`, gated on `CurrentBlockLifted`),
   skip the redundant edge, and drain the stale logical-stack copy
   (`evm_bytecode_visitor.h`).

3. **Operand identity key on a non-U256 operand** — `getOperandIdentityKey`
   dereferenced `getU256VarComponents()` on a compile-time type check alone, so
   an empty/deferred operand hit the "Not a multi-component U256" assert. Add a
   runtime `isU256MultiComponent()` guard (`evm_lifted_stack_lifter.h`).

## Validation

| Suite | Flag | Result |
|---|---|---|
| multipass `evmone-unittests` | ON | 223/223 |
| multipass `evmone-statetest -k fork_Cancun` (EEST) | ON | **2723/2723**, 0 failed, no SIGABRT |
| 227-fixture mainnet-replay corpus | ON | no SIGABRT; fail-set identical to OFF (0 regressions) |
| multipass `evmone-unittests` | OFF | 223/223 |
| multipass `evmone-statetest -k fork_Cancun` | OFF | 2723/2723 |
| `tools/format.sh check` | — | passes |

## Known residual (not a crash)

The EEST run still emits five `PredBlockPC == BlockPC` self-edge cases on
single-predecessor blocks (no merge phis, no abort, all tests pass). Clearing
them would need a broader audit of the terminator/JUMPDEST begin handshake
rather than a surgical change.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/compiler/evm_frontend/evm_analyzer.h [hunk 1]
```diff
@@ -250,6 +250,23 @@ class EVMAnalyzer {
...
+      return -1;
+    }
+    for (const auto &[RegionEntryPC, RegionInfo] : DynamicJumpRegions) {
+      (void)RegionEntryPC;
+      if (RegionInfo.ShapeClassId == ShapeClassId) {
+        return RegionInfo.FullEntryStateDepth;
+      }
+    }
+    return -1;
+  }
```
[the remark this answers is not in the document]
Comment (abmcar) [reply src/compiler/evm_frontend/evm_analyzer.h:263]: Done in ae9a670 — the ShapeClassId → entry-depth mapping is now populated alongside ShapeClassId assignment in `finalizeDynamicJumpRegionMetadata` (an `unordered_map`), and `getDynamicJumpShapeClassEntryDepth` is a direct O(1) lookup.
